dashboard/app/controllers/user_school_infos_controller.rb

```diff
@@ -0,0 +1,57 @@
+class UserSchoolInfosController < ApplicationController
+  # PATCH /api/v1/users_school_infos/<id>/update_last_confirmation_date
+  def update_last_confirmation_date
+    user_school_info = UserSchoolInfo.find(params[:id])
+    result = user_school_info.update ({
+      last_confirmation_date: DateTime.now
+    })
+    if result
+      head :no_content
+    else
+      render json: user_school_info.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH /api/v1/users_school_infos/<id>/update_end_date_last_seen_school_info_interstitial
+  def update_end_date_last_seen_school_info_interstitial
+    user_school_info = UserSchoolInfo.find(params[:id])
+
+    user_school_info.update!(end_date: DateTime.now)
```

Comment:
two things here:
1. lines 19 and 21 should be inside a transaction so we can rollback if one command fails
2. if a command fails, how should we respond?

Reply:
Good question.  I have modified the method to update the end_date first before updating the last school info interstitial.  I utilized a call back in the model to accomplish this.  Will this suffice?  Or is it better to wrap both updates in a transaction?

Reply:
i think it's safer to wrap both updates in a transaction here. it's not a _huge_ deal if one command fails, but it's also a little clearer (in terms of intention) to have both commands in one place